impl/src/test/java/org/ehcache/loaderwriter/writebehind/AbstractWriteBehindTestBase.java

```diff
@@ -592,4 +594,40 @@ public void testAgedBatchedIsWritten() throws Exception {
       cacheManager.close();
     }
   }
+
+  @Test
+  public void testWriteBehindQueueSize() throws Exception {
+    
+    class TestWriteBehindProvider extends WriteBehindProviderFactory.Provider {
+
+      private WriteBehind writeBehind = null;
+
+      @Override
+      public <K, V> WriteBehind<K, V> createWriteBehindLoaderWriter(final CacheLoaderWriter<K, V> cacheLoaderWriter, final WriteBehindConfiguration configuration) {
+        this.writeBehind = super.createWriteBehindLoaderWriter(cacheLoaderWriter, configuration);
+        return writeBehind;
+      }
+
+      public WriteBehind getWriteBehind() {
+        return writeBehind;
+      }
+    }
+
+    TestWriteBehindProvider writeBehindProvider = new TestWriteBehindProvider();
+    WriteBehindTestLoaderWriter<String, String> loaderWriter = new WriteBehindTestLoaderWriter<String, String>();
+
+    CacheManager cacheManager = managerBuilder().using(writeBehindProvider).build(true);
+    try {
+      Cache<String, String> testCache = cacheManager.createCache("testAgedBatchedIsWritten", configurationBuilder()
+          .add(new DefaultCacheLoaderWriterConfiguration(loaderWriter))
+          .add(newBatchedWriteBehindConfiguration(5, SECONDS, 2).build())
```

Comment:
This test will run for 5 seconds+ if left like that, do complete the batch after your assertion to make it finish faster